packages/aws-cdk-lib/aws-stepfunctions/lib/activity.ts

```diff
@@ -15,6 +17,26 @@ export interface ActivityProps {
    * @default - If not supplied, a name is generated
    */
   readonly activityName?: string;
+
+  /**
+   * Specifies a symmetric customer managed KMS key for server-side encryption of the activity inputs.
+   * Step Functions will reuse the key for a maximum of `kmsDataKeyReusePeriodSeconds`.
+   *
+   * @default - data is transparently encrypted using an AWS owned key
+   */
+  readonly kmsKey?: kms.IKey;
+
+  /**
+   * Maximum duration that Step Functions will reuse customer managed data keys.
+   * When the period expires, Step Functions will call GenerateDataKey.
```

Comment:
What happens when `GenerateDataKey` is called?

Reply:
AWS KMS returns the plaintext data key in the response, and each SFN execution will use the data key to encrypt execution inputs/outputs, and events.